accepted/0000-workspaces.md

```diff
@@ -0,0 +1,479 @@
+        "lodash": "^4.x.x"
+    }
+}
+
+$ cat ./packages/workspace-a/package.json
+{
+    "name": "workspace-a",
+    "version": "1.7.3",
+    "peerDependencies": {
+        "react": "^16.x.x"
+    },
+    "dependencies": {
+        "workspace-b": "^2.0.0"
+    }
+}
+
+$ cat ./packages/worskpace-b/package.json
+{
+    "name": "workspace-b",
+    "version": "2.1.1",
+    "peerDependencies": {
+        "react": "^16.x.x"
+    }
+}
+```
+
+Will result in the following symlinking structure:
+
+```
+$ tree
+.
+├── package-lock.json
+├── node_modules
+│   ├── lodash
+│   ├── libnpmutil -> ./core/libnpmutil
+│   ├── workspace-a -> ./packages/workspace-a
+│   ├── workspace-b -> ./packages/workspace-b
+│   └── react
+├── core
+│   └── libnpmutil
+│       ├── package-lock.json
+│       └── node_modules
+│           └── lodash -> ../../../node_modules/lodash
+└── packages
+    ├── workspace-a
+    │   ├── package-lock.json
+    │   └── node_modules
+    │       ├── react -> ../../../node_modules/react
+    │       └── worspace-b -> ../../../node_modules/workspace-b
+    └─ worspace-b
+        ├── package-lock.json
+        └── node_modules
+            └── react -> ../../../node_modules/react
+```
+
+And the following `package-lock.json` files:
+
+```
+$ cat ./package-lock.json
+{
+  "name": "foo",
+  "version": "1.0.0",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {
+    "": {
+      "name": "foo",
+      "version": "1.0.0",
+      "dependencies": {
+        "lodash": "^4.17.15",
+        "libnpmutil": "^1.0.0"
+      }
+    },
+    "core/libnpmutil": {
+      "name": "libnpmutil",
+      "version": "1.0.0"
+    },
+    "node_modules/lodash": {
+      "name": "lodash",
+      "version": "4.17.15",
+      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.15.tgz",
+      "integrity": "sha512-8xOcRHvCjnocdS5cpwXQXVzmmh5e5+saE2QGoeQmbKmRS6J3VQppPOIt0MnmE+4xlZoumy0GPG0D0MVIQbNA1A=="
+    },
+    "node_modules/libnpmutil": {
+      "resolved": "core/libnpmutil",
+      "link": true
+    },
+    "node_modules/workspace-a": {
+      "resolved": "packages/workspace-a",
+      "link": true
+    },
+    "node_modules/workspace-b": {
+      "resolved": "packages/workspace-b",
+      "link": true
+    },
+    "packages/workspace-a": {
+      "name": "workspace-a",
+      "version": "1.7.3"
+    },
+    "packages/workspace-b": {
+      "name": "workspace-b",
+      "version": "2.1.1"
+    }
+  },
+  "dependencies": {
+    "lodash": {
+      "version": "4.17.15",
+      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.15.tgz",
+      "integrity": "sha512-8xOcRHvCjnocdS5cpwXQXVzmmh5e5+saE2QGoeQmbKmRS6J3VQppPOIt0MnmE+4xlZoumy0GPG0D0MVIQbNA1A=="
+    },
+    "libnpmutil": {
+      "version": "file:core/libnpmutil"
+    },
+    "workspace-a": {
+      "version": "file:packages/workspace-a"
+    },
+    "workspace-b": {
+      "version": "file:packages/workspace-b"
+    }
+  }
+}
+
+$ cat ./packages/workspace-a/package-lock.json
```

Comment:
If the root lockfile already serializes all of this information, why are the nested lockfiles necessary? Or am I missing something? (I'm probably missing something)

Or, looking closer, I see that the nested lockfiles _are_ specifying their "local" dependencies etc. Why can't that information be hoisted to the root lockfile's `packages` subtree?

Reply:
The thinking here is that you might define a workspace with a bunch of other packages (as separate git working dirs), and you'd still want those other projects to have a lockfile included for users who check them out independently.

Reply:
as discussed in the [OpenRFC Deep Dive call](https://github.com/npm/rfcs/issues/106) this is no longer going to be the case - RFC have been updated to reflect this change.